code/modules/vending/runic_vendor.dm

```diff
@@ -60,8 +60,7 @@
 
 	return .
 
-
-/obj/machinery/vending/runic_vendor/Destroy()
+/obj/machinery/vending/runic_vendor/on_deconstruction(disassembled)
```

Comment:
Subtle bug here, but when a runic vendor "decays" it just `qdel(src)`. That means this code won't run when it's meant to. 

You should replace the `qdel(src)`s with `deconstruct(FALSE)`s. And also give the machine `NO_DEBRIS_ON_DECONSTRUCTION` if it doesn't already have it.

Reply:
Nice catch, it also looks like I have to do the same in its `screwdriver_act`.

Reply:
If you are using `NO_DEBRIS_AFTER_DECONSTRUCTION` then you need to instead override `handle_deconstruct()` because the default behaviour of this flag for machines is to not call `on_deconstruction()` all together.

`handle_deconstruct()` as the name implies allows you to control all aspects of deconstruction including how you want to handle the `NO_DEBRIS_AFTER_DECONSTRUCTION` flag

You would do something like this

```
/obj/machinery/vending/runic_vendor/handle_deconstruct(disassembled)
   SHOULD_NOT_OVERRIDE(FALSE)

  visible_message(span_warning("[src] flickers and disappears!"))
  playsound(src,'sound/weapons/resonator_blast.ogg',25,TRUE)
  return ..()
```

Reply:
If you are using NO_DEBRIS_AFTER_DECONSTRUCTION then you need to instead override handle_deconstruct() because the default behaviour of this flag for machines is to not call on_deconstruction() all together.

handle_deconstruct() as the name implies allows you to control all aspects of deconstruction including how you want to handle the NO_DEBRIS_AFTER_DECONSTRUCTION flag

You would do something like this

/obj/machinery/vending/runic_vendor/handle_deconstruct(disassembled)
   SHOULD_NOT_OVERRIDE(TRUE)

  visible_message(span_warning("[src] flickers and disappears!"))
  playsound(src,'sound/weapons/resonator_blast.ogg',25,TRUE)
  return ..()

Reply:
Thank you! I wouldn't have dared to override `hand_deconstruct` on my own haha.